tests/test_backend.py

```diff
@@ -31,174 +31,166 @@ def reverse_key(key):
     return key.split("#", 2)[2]
 
 
-class DjangoRedisConnectionStrings(unittest.TestCase):
-    def test_connection_strings(self):
-        connection_strings = [
-            "unix://tmp/foo.bar?db=1",
-            "redis://localhost/2",
-            "rediss://localhost:3333?db=2",
-        ]
-        cf = pool.get_connection_factory(
-            path="django_redis.pool.ConnectionFactory", options={}
-        )
-        for connection_string in connection_strings:
-            with self.subTest(connection_string):
-                res = cf.make_connection_params(connection_string)
-                self.assertEqual(res["url"], connection_string)
-
-
-class DjangoRedisCacheTestEscapePrefix(unittest.TestCase):
-    def setUp(self):
-
-        caches_setting = copy.deepcopy(settings.CACHES)
-        caches_setting["default"]["KEY_PREFIX"] = "*"
-        cm = override_settings(CACHES=caches_setting)
-        cm.enable()
-        self.addCleanup(cm.disable)
-
-        self.cache = caches["default"]
-        self.other = caches["with_prefix"]
-
-    def tearDown(self):
-        self.cache.clear()
-        self.other.clear()
-
-    def test_delete_pattern(self):
-        self.cache.set("a", "1")
-        self.other.set("b", "2")
-        self.cache.delete_pattern("*")
-        self.assertIs(self.cache.has_key("a"), False)
-        self.assertEqual(self.other.get("b"), "2")
-
-    def test_iter_keys(self):
-        if isinstance(self.cache.client, ShardClient):
-            self.skipTest("ShardClient doesn't support iter_keys")
-
-        self.cache.set("a", "1")
-        self.other.set("b", "2")
-        self.assertEqual(list(self.cache.iter_keys("*")), ["a"])
-
-    def test_keys(self):
-        self.cache.set("a", "1")
-        self.other.set("b", "2")
-        keys = self.cache.keys("*")
-        self.assertIn("a", keys)
-        self.assertNotIn("b", keys)
+@pytest.mark.parametrize(
+    "connection_string",
+    [
+        "unix://tmp/foo.bar?db=1",
+        "redis://localhost/2",
+        "rediss://localhost:3333?db=2",
+    ],
+)
+def test_connection_strings(connection_string):
+    cf = pool.get_connection_factory(
+        path="django_redis.pool.ConnectionFactory", options={}
+    )
+    res = cf.make_connection_params(connection_string)
+    assert res["url"] == connection_string
+
+
+@pytest.fixture()
+def available_caches(settings):
```

Comment:
could you add a little bit of type hints? and maybe also separate them? just so that it looks cleaner and you do not have to do the unpacking in every test